docs/learn/encyclopedia/transactions-specialized/channel-accounts.mdx

```diff
@@ -643,94 +636,174 @@ Here's an example of what that might looks like:
-function markChannelAccountAvailable(channelAccount) {
-  channelAccount.state = "available";
+List<Transaction> txOutput = new ArrayList<>();
+int recipientIndex = 0;
+
+// Build transactions dynamically based on available recipients
+while (recipientIndex < allRecipients.length) {
+    for (ChannelData channelData : channelAccountsTracker) {
+        if (recipientIndex >= allRecipients.length) {
+            break;
+        }
+        if (channelData.state.equals("available")) {
+            channelData.state = "active";
+            Transaction.Builder txBundleBuilder = new Transaction.Builder(channelData.account, Network.TESTNET)
+                    .setBaseFee(640)
+                    .setTimeout(3600);
+
+            // Add up to 100 payment operations per channel
+            for (int j = 0; j < 100; j++) {
+                if (recipientIndex >= allRecipients.length) {
+                    break;
+                }
+                PaymentOperation paymentOperation = new PaymentOperation.Builder(
+                        allRecipients[recipientIndex],
+                        AssetTypeNative.INSTANCE,
+                        "10"
+                ).setSourceAccount(primaryKeypair.getAccountId()).build();
+                txBundleBuilder.addOperation(paymentOperation);
+                recipientIndex++;
+            }
+
+            Transaction txBundle = txBundleBuilder.build();
+            txBundle.sign(primaryKeypair);
+            txBundle.sign(channelData.keypair);
+
+            txOutput.add(txBundle);
+        } else {
+            System.out.println("No available channel account. Waiting...");
+            // Perhaps sleep here to wait for availability
+        }
+    }
 }
 
-// Send payments using channel accounts with state rotation
-(async () => {
-  const server = new StellarSdk.Server("https://horizon-testnet.stellar.org");
-  const sourceKeypair = StellarSdk.Keypair.fromSecret(
-    "SDZ4Z5K7UQW4RQD5K6ZP3W6ZKHME4GSS2BGSRWLU3AFB4MPRU5Q5MRBJ",
-  );
-  const sourceAccount = await server.loadAccount(sourceKeypair.publicKey());
-
-  let n = 0; // Transaction increment counter
-  for (let i = 0; i < recipientKeys.length; i += 100) {
-    const channelAccount = getAvailableChannelAccount();
-    if (!channelAccount) {
-      console.log(
-        "No available channel account found. Waiting for one to become available...",
-      );
-      await new Promise((resolve) => setTimeout(resolve, 1000));
-      i -= 100; // Retry this batch
-      continue;
+List<String> failedRecipients = new ArrayList<>();
+
+// Submit transactions
+for (Transaction bundle : txOutput) {
+    ChannelData channelData = getChannelDataForTransaction(txOutput, bundle, channelAccountsTracker);
+    if (channelData != null) {
+        try {
+            SubmitTransactionResponse response = server.submitTransaction(bundle);
+            System.out.println("Transaction succeeded with hash: " + response.getHash());
+        } catch (Exception e) {
```

Comment:
Where is the System namespace

Reply:
This ought be related to _[Id.](https://github.com/stellar/stellar-docs/pull/723#discussion_r1811560112)_ and working now since it's my understanding that the `System` class is part of the `java.lang` package, which is automatically imported by default in every Java program. 🧑‍💻 GPT says this means you do not need to explicitly import or configure the System class to use `System.out.println()` or any other methods thereto. 🤔